framework/src/modules/token/cc_method.ts

```diff
@@ -86,37 +86,61 @@ export class TokenInteroperableMethod extends BaseCCMethod {
 	}
 
 	public async beforeCrossChainMessageForwarding(ctx: CrossChainMessageContext): Promise<void> {
-		const { ccm } = ctx;
+		const {
+			transaction: { senderAddress: relayerAddress },
+			ccm,
+		} = ctx;
+		const { ccmID } = getEncodedCCMAndID(ccm);
 		const methodContext = ctx.getMethodContext();
 		const messageFeeTokenID = await this._interopMethod.getMessageFeeTokenID(
 			methodContext,
 			ccm.receivingChainID,
 		);
-		const { ccmID } = getEncodedCCMAndID(ccm);
-
 		const escrowStore = this.stores.get(EscrowStore);
-		const escrowKey = escrowStore.getKey(ccm.sendingChainID, messageFeeTokenID);
-		const escrowAccount = await escrowStore.getOrDefault(methodContext, escrowKey);
-		if (escrowAccount.amount < ccm.fee) {
-			this.events.get(BeforeCCMForwardingEvent).error(
-				methodContext,
-				ccm.sendingChainID,
-				ccm.receivingChainID,
-				{
-					ccmID,
-					messageFeeTokenID,
-				},
-				TokenEventResult.INSUFFICIENT_ESCROW_BALANCE,
-			);
+		const sendingChainEscrowKey = escrowStore.getKey(ccm.sendingChainID, messageFeeTokenID);
+		const sendingChainEscrowAccount = await escrowStore.getOrDefault(
+			methodContext,
+			sendingChainEscrowKey,
+		);
+		if (ccm.fee > 0) {
+			if (sendingChainEscrowAccount.amount < ccm.fee) {
+				this.events.get(BeforeCCMForwardingEvent).error(
+					methodContext,
+					ccm.sendingChainID,
+					ccm.receivingChainID,
+					{
+						ccmID,
+						messageFeeTokenID,
+					},
+					TokenEventResult.INSUFFICIENT_ESCROW_BALANCE,
+				);
 
-			throw new Error('Insufficient balance in the sending chain for the message fee.');
+				throw new Error('Insufficient balance in the sending chain for the message fee.');
+			}
+			// Deduct the fee from escrow of the sending chain.
+			sendingChainEscrowAccount.amount -= ccm.fee;
+			await escrowStore.set(methodContext, sendingChainEscrowKey, sendingChainEscrowAccount);
+
+			// If the ccm will get forwarded, we update the escrow of the receiving chain.
+			if (ccm.status !== CCMStatusCode.FAILED_CCM) {
```

Comment:
its not the ccm.status but its coming from the context, `ctx.ccmFailed` and still in PR https://github.com/LiskHQ/lisk-sdk/pull/8817/files#diff-a84902ce0adabb5f76267d84f856776806efd075eefb7854d66513f11077a810, also update the tests accordingly

Reply:
Added the typing from the linked PR and updated the tests